src/hotspot/share/code/codeCache.cpp

```diff
@@ -664,8 +664,8 @@ CodeBlob* CodeCache::find_blob(void* start) {
 
 nmethod* CodeCache::find_nmethod(void* start) {
   CodeBlob* cb = find_blob(start);
-  assert(cb != nullptr, "did not find an nmethod");
-  return cb->as_nmethod();
+  assert(cb == nullptr || cb->is_nmethod(), "did not find an nmethod");
+  return (nmethod*)cb;
```

Comment:
There's a call to `find_nmethod` in `ZNMethod::load_oop` that now lacks a null-check. Would you mind adding one?

Reply:
I added asserts in all places where there is no nullptr check.